server/routes/users.js

```diff
@@ -4,17 +4,26 @@ const router = express.Router()
 const { User } = require( '../database' )
 
 router.post( '/sign-up', ( request, response ) => {
+  //if request.session.user, don't even let someone see this page
   const { username, email, password, lat, long } = request.body
 
   User.signUp({ username, email, password, lat, long })
     .then( user => response.status( 201 ).json( user ))
     .catch( error => response.status( 500 ).json({ error: "The email or username you entered were already taken." }))
+    //TODO: add handling (maybe on front end?) if there is missing info. that creates an error
```

Comment:
Should have a check on the front end and back end. The check on the front end is good for users but if you access the route through postman or something that won't be enough to block it.

Reply:
Thank you!